Sources/TSCBasic/Path.swift

```diff
@@ -370,6 +373,10 @@ private struct PathImpl: Hashable {
     /// string consisting of just `.` if there is no directory part (which is
     /// the case if and only if there is no path separator).
     fileprivate var dirname: String {
+#if USE_FOUNDATION
```

Comment:
I don't think such a define makes sense since SwiftPM is not trying to avoid Foundation. We just want better APIs for our use case. I am perfectly fine with falling back to Foundation for Windows though. So maybe change this to `#if os(windows)`?

Reply:
Since Swift-Foundation does essentially the same thing this does ([using `String.prefix(upto:...)`](https://github.com/apple/swift-corelibs-foundation/blob/master/Foundation/NSPathUtilities.swift#L241)), I believe this is the path it should take on Linux as well? If I recall correctly, the main reason for wanting to avoid foundation is the cost of bridging into objc on macOS.

Reply:
Yeah, bridging cost is also factor for not using Foundation in some of the performance sensitive areas.